product_attribute_value_auto_fill_option/models/product_template_attribute_line.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2025 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductAttribute(models.Model):
+    _inherit = "product.template.attribute.line"
+
+    @api.onchange("attribute_id")
+    def _onchange_attribute_id(self):
+        # Select all values just when the option 'fill_all_values' on
+        # the attribute is selected.
+        if self.attribute_id.fill_all_values:
+            self.value_ids = self.env["product.attribute.value"].search(
+                [
+                    ("attribute_id", "=", self.attribute_id.id),
+                ]
+            )
+        elif (
+            not self.attribute_id.fill_all_values
+            and self.attribute_id.create_variant == "no_variant"
+        ):
+            self.value_ids = self.value_ids.filtered(
+                lambda pav: pav.attribute_id == self.attribute_id
+            )
+        else:
+            return super()._onchange_attribute_id()
```

Comment:
I think we could simplify it like this:
```suggestion
            return super()._onchange_attribute_id()
        self.value_ids = False
```